### PR TITLE
[grafana] Dashboard linter

### LINKED ISSUE #7081: [grafana] Dashboard linter
CI must fail when:

- Dashboard contains a deprecated query interval (deprecated intervals: `$__interval_rv`, `$__interval_sx3` and `$__interval_sx4`)
- Dashboard contains a legacy alert rule (alert rule that is configured on dashboard table and uses Grafana < 9.x internal alerting)
- Dashboard contains a hard-coded datasource uid (uid is copied as-is, without using Grafana variable of `Datasource` type)

### PR DESCRIPTION
**Pull request by ghostinsoba**
## Description
Grafana dashboard linter

## Why do we need it, and what problem does it solve?
Closes #7081
Related to #7018 and #7027

## What is the expected result?
CI fail when:

- Dashboard contains a deprecated query interval (deprecated intervals: `$__interval_rv`, `$__interval_sx3` and `$__interval_sx4`)
- Dashboard contains a legacy alert rule (alert rule that is configured on dashboard table and uses Grafana < 9.x internal alerting)
- Dashboard contains a hard-coded datasource uid (uid is copied as-is, without using Grafana variable of `Datasource` type)

## Checklist
- [x] The code is covered by unit tests.
- [ ] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [ ] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: tools | testing
type: chore
summary: Grafana dashboard linter
impact_level: low
```

### CODE REVIEW

File: tools/validation/grafana_dashboard.go [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/tidwall/gjson"
+)
+
+func RunGrafanaDashboardValidation(info *DiffInfo) (exitCode int) {
+	fmt.Printf("Run 'grafana dashboards' validation ...\n")
```
Comment (name212) [inline tools/validation/grafana_dashboard.go:28]:
```suggestion
	fmt.Println("Run 'grafana dashboards' validation ...")
```

File: tools/validation/grafana_dashboard.go [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+
+	if len(info.Files) == 0 {
+		fmt.Printf("Nothing to validate, diff is empty\n")
```
Comment (name212) [inline tools/validation/grafana_dashboard.go:31]:
```suggestion
		fmt.Println("Nothing to validate, diff is empty.")
```

File: tools/validation/grafana_dashboard.go [hunk 3]
```diff
@@ -0,0 +1,172 @@
+/*
+Copyright 2021 Flant JSC
```
Comment (name212) [inline tools/validation/grafana_dashboard.go:2]:
```suggestion
Copyright 2024 Flant JSC
```

File: testing/validate_grafana_dashboard.sh [hunk 4]
```diff
@@ -0,0 +1,21 @@
+#!/bin/bash
+
+# Copyright 2021 Flant JSC
```
Comment (name212) [inline testing/validate_grafana_dashboard.sh:3]:
```suggestion
# Copyright 2024 Flant JSC
```

File: tools/validation/grafana_dashboard_test.go [hunk 5]
```diff
@@ -0,0 +1,391 @@
...
+*/
+
+package main
+
+import (
+	"reflect"
+	"testing"
+)
+
+func Test_validateGrafanaDashboardFile(t *testing.T) {
```
Comment (name212) [inline tools/validation/grafana_dashboard_test.go:24]:
```suggestion
func TestValidateGrafanaDashboardFile(t *testing.T) {
```

File: tools/validation/grafana_dashboard.go (hunk 1 above)
Comment (ghostinsoba) [reply tools/validation/grafana_dashboard.go]: fixed

File: tools/validation/grafana_dashboard.go (hunk 2 above)
Comment (ghostinsoba) [reply tools/validation/grafana_dashboard.go]: fixed

File: tools/validation/grafana_dashboard.go [hunk 6]
```diff
@@ -0,0 +1,172 @@
...
+		exitCode = 1
+	}
+
+	return exitCode
+}
+
+func isGrafanaDashboard(fileName string) bool {
+	fileName = strings.ToLower(fileName)
+	return strings.Contains(fileName, "grafana") &&
+		strings.Contains(fileName, "dashboard") &&
```
[the remark this answers is not in the document]
Comment (ghostinsoba) [reply tools/validation/grafana_dashboard.go]: fixed

File: tools/validation/grafana_dashboard.go (hunk 3 above)
Comment (ghostinsoba) [reply tools/validation/grafana_dashboard.go]: fixed

File: testing/validate_grafana_dashboard.sh (hunk 4 above)
Comment (ghostinsoba) [reply testing/validate_grafana_dashboard.sh]: fixed

File: tools/validation/grafana_dashboard_test.go (hunk 5 above)
Comment (ghostinsoba) [reply tools/validation/grafana_dashboard_test.go]: fixed

**Review by name212 — APPROVED** [review]

**Pull request merged**